test/crypto_subtle_test.dart

```diff
@@ -0,0 +1,311 @@
+          isA<UnsupportedError>(),
+        ),
+      );
+    });
+  });
+
+  group('crypto', () {
+    test('getRandomValues: success', () {
+      final data = Uint8List(16 * 1024);
+      expect(
+        data.every((e) => e == 0),
+        isTrue,
+      );
+      subtle.window.crypto.getRandomValues(data.toJS);
+      expect(
+        data.any((e) => e != 0),
+        isTrue,
+      );
+    });
+
+    test('getRandomValues: too long', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Uint8List(1000000).toJS),
+        throwsA(
+          isA<subtle.JSDomException>().having(
+            (e) => e.name,
+            'name',
+            'QuotaExceededError',
+          ),
+        ),
+      );
+    });
+
+    test('getRandomValues: not supported type', () {
+      expect(
+        () => subtle.window.crypto.getRandomValues(Float32List(32).toJS),
+        throwsA(
+          isA<subtle.JSDomException>().having(
+            (e) => e.name,
+            'name',
+            'TypeMismatchError',
+          ),
+        ),
+      );
+    });
+  });
+
+  group('crypto.subtle', () {
+    test('generateCryptoKey: success', () async {
+      expect(
+        await subtle.window.crypto.subtle
+            .generateCryptoKey(
+              const subtle.Algorithm(
+                name: 'AES-GCM',
+                length: 256,
+              ).toJS,
+              false,
+              ['encrypt', 'decrypt'].toJS,
+            )
+            .toDart,
+        isA<subtle.JSCryptoKey>()
+            .having(
+              (key) => key.type,
+              'type',
+              'secret',
+            )
+            .having(
+              (key) => key.extractable,
+              'extractable',
+              false,
+            )
+            .having(
+              (key) => key.usages,
+              'usages',
+              containsAll(['encrypt', 'decrypt']),
+            ),
+      );
+    });
+
+    test('generateCryptoKey: invalid keyUsages', () {
+      expect(
+        () async => await subtle.window.crypto.subtle
+            .generateCryptoKey(
+              const subtle.Algorithm(
+                name: 'AES-GCM',
+                length: 256,
+              ).toJS,
+              false,
+              <String>[].toJS,
+            )
+            .toDart,
+        throwsA(
+          isA<subtle.JSDomException>().having(
+            (e) => e.name,
+            'name',
+            'SyntaxError',
+          ),
+        ),
+      );
+    });
+
+    test(testOn: 'chrome || safari', 'generateCryptoKey: invalid algorithm',
+        () {
+      expect(
+        () async => await subtle.window.crypto.subtle
+            .generateCryptoKey(
+              const subtle.Algorithm().toJS,
+              false,
+              ['encrypt', 'decrypt'].toJS,
+            )
+            .toDart,
+        throwsA(
+          isA<subtle.JSDomException>().having(
+            (e) => e.name,
+            'name',
+            'TypeError',
```

Comment:
nit:
```suggestion
            // Chrome / Safari throw TypeError
            // Firefox throws SyntaxError
            anyOf('TypeError', 'SyntaxError'),
```
Testing that it's either a `TypeError` or `SyntaxError` would also be fine.

Reply:
Thank you! I thought it was good!